libs/execution_context/ebpf_native.c

```diff
@@ -1233,7 +1239,7 @@ ebpf_native_load_programs(
         goto Done;
     }
 
-    if (module->unloading || module->detaching) {
+    if (module->detaching || module->state == MODULE_STATE_UNLOADING) {
```

Comment:
As earlier, I think we can merge this if check and the one at line 1231.

Reply:
@saxena-anurag Sure fine with that, just want to ensure we really want to merge the current distinction between `EBPF_OBJECT_ALREADY_EXISTS` and `EBPF_EXTENSION_FAILED_TO_LOAD` (should any existing code rely on this).
If so, what unified error would be more appropriate?